src/components/todo/TodoList.tsx

```diff
@@ -7,11 +7,11 @@ export interface Props {
   setTodos: React.Dispatch<React.SetStateAction<TodoModal[]>>; // copied from setTodos state
 }
 
-const TodoList = ({ todos }: Props) => {
+const TodoList = ({ todos, setTodos }: Props) => {
   return (
     <div>
-      {todos.map((allTodo) => (
-        <SingleTodo todoSingleObj={allTodo} />
+      {todos.map((allTodo, index) => (
+        <SingleTodo todoArrayOfObj={todos} setTodos={setTodos} index={index} />
```

Comment:
Can we re-think about this.
I would hint few items.

1. **SingleTodo** doesn't sound something, can it be **TodoItem**?
2. Does **SingleTodo** requires all the todo items?
3. **todoArrayOfObj** name sounds like **DumbleDoor** from HP.
4. I believe every todo item will have a unique id.

Reply:
1. SingleTodo renamed as TodoItem - 
2. Yes, todoItems (previously it was SingleTodo) requires all the todo and we need that to filter out "task done" items
3. todoArrayOfObj  renamed as "todoItems"
4. **I believe every todo item will have a unique id**. - I think you asked this question because I am also passing "index" in  "TodoItem" component, so yes every todo item is having unique id, but still we need index to filter out "task done" items.

Kindly confirm if I am not in the right direction.

Reply:
We can use Array.find() by ID, both approach will work.